plugins/inputs/logfile/logfile.go

```diff
@@ -35,14 +37,22 @@ type LogFile struct {
 	configs           map[*FileConfig]map[string]*tailerSrc
 	done              chan struct{}
 	removeTailerSrcCh chan *tailerSrc
+	tailerSemaphore   *semaphore.Semaphore
 	started           bool
 }
 
 func NewLogFile() *LogFile {
+	fileDescriptorLimit, err := fdlimit.CurrentFileDescriptorLimit()
+
+	if err != nil {
+		log.Panicf("E! Failed to get file descriptor limit:", err)
+	}
```

Comment:
No. As you have said, the file descriptors are only useful for us instead of customers. Therefore, this feature should be non-intrusive to customer side by utilizing 
```
return &LogFile{
			numUsedFds:   	   semaphore.NewSemaphore(0),
}
```

and 

```
if t.numUsedFds.GetLimit != 0 {
		t.Log.Debugf("Number of file descriptors used by agent / Total allowed file descriptors: %v/%v", t.numUsedFds.GetCount(), t.numUsedFds.GetLimit())
}
```